llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -17534,10 +17534,13 @@ template <class MatchContextClass> SDValue DAGCombiner::visitFMA(SDNode *N) {
       return N2;
   }
 
+  const bool PreferFMAAdd = (TLI.isOperationLegal(ISD::FMA, VT) &&
```

Comment:
Does the legality of `ISD::FMA x VT` actually matter?

Reply:
Yes, in cases where both fma and add are illegal then I don't see any reason not to simplify the fma to an add. The subsequent legalization of fadd may result in a better sequence of instructions than the fma legalization.